packages/eslint-plugin/src/rules/no-var-requires.ts

```diff
@@ -32,6 +33,8 @@ export default util.createRule<Options, MessageIds>({
             ? node.parent.parent
             : node.parent;
 
+        const variable = ASTUtils.findVariable(context.getScope(), 'require');
```

Comment:
small optimization, can you move this search to only be executed if constant conditions do pass?

```
          // constant checks
          parent &&
          [
            AST_NODE_TYPES.CallExpression,
            AST_NODE_TYPES.MemberExpression,
            AST_NODE_TYPES.NewExpression,
            AST_NODE_TYPES.TSAsExpression,
            AST_NODE_TYPES.TSTypeAssertion,
            AST_NODE_TYPES.VariableDeclarator,
          ].includes(parent.type)
```

Reply:
```ts
        if (
          !parent ||
          ![
            AST_NODE_TYPES.CallExpression,
            AST_NODE_TYPES.MemberExpression,
            AST_NODE_TYPES.NewExpression,
            AST_NODE_TYPES.TSAsExpression,
            AST_NODE_TYPES.TSTypeAssertion,
            AST_NODE_TYPES.VariableDeclarator,
          ].includes(parent.type)
        ) {
          return;
        }

        const variable = ASTUtils.findVariable(context.getScope(), 'require');
        if (!variable?.identifiers.length) {
          return;
        }
         
        context.report({
          node,
          messageId: 'noVarReqs',
        });
```

or

```ts
        if (
          parent &&
          [
            AST_NODE_TYPES.CallExpression,
            AST_NODE_TYPES.MemberExpression,
            AST_NODE_TYPES.NewExpression,
            AST_NODE_TYPES.TSAsExpression,
            AST_NODE_TYPES.TSTypeAssertion,
            AST_NODE_TYPES.VariableDeclarator,
          ].includes(parent.type)
        ) {
          const variable = ASTUtils.findVariable(context.getScope(), 'require');
          if (variable?.identifiers.length) {
            context.report({
              node,
              messageId: 'noVarReqs',
            });
          }
        }
```

Reply:
Done. BTW the condition in your first snippet is incorrect. 
```Typescript
         // shouldn't have ! here since we want to return if variable?.identifiers.length is truthy
        if (!variable?.identifiers.length) {
          return;
        }
```